### PR TITLE
Add local catalog path to .env

### PR DESCRIPTION
**Pull request by bnchrch**
## What

We have a way to override the path to the local catalog in oss development.

However there are two issues
1. its not currently documented and is not visible to the average user.
2. It didn't have support for docker-compose

## How
1. This adds the .env.dev reference
2. And also adds the env var to docker-compose.yml

### CODE REVIEW

File: airbyte-config/init/src/main/java/io/airbyte/config/init/LocalDefinitionsProvider.java
Comment (pedroslopez) [inline airbyte-config/init/src/main/java/io/airbyte/config/init/LocalDefinitionsProvider.java:37]: nit: add the word `connector` in there

Comment (pedroslopez) [reply airbyte-config/init/src/main/java/io/airbyte/config/init/LocalDefinitionsProvider.java:37]: also i'm not sure how often this will be logged - should probably only happen on initialization if possible

**Review by pedroslopez — APPROVED** [review]

File: airbyte-config/config-models/src/main/java/io/airbyte/config/CatalogDefinitionsConfig.java
Comment (pedroslopez) [inline airbyte-config/config-models/src/main/java/io/airbyte/config/CatalogDefinitionsConfig.java:17]: nit: this should have used micronaut configs and specified the env var via the application.yml, instead of EnvConfigs, which will go away soon.

File: docker-compose.yaml
Comment (pedroslopez) [inline docker-compose.yaml:31]: Do we need to add this to any of the other apps? I think the bootloader is the only place where this is used because that's where we seed things, but just checking in case we use it in the server or elsewhere as well for some reason (i believe the server should just be using whatever is in the db rather than the local provider)

**Review by evantahler — APPROVED** [review]
👍 from me on the DX, but I defer my technical review to the others.

File: docker-compose.yaml
Comment (bnchrch) [reply docker-compose.yaml:31]: I tested around and it seems like its only the boot loader that requires it

**Comment by prateekmukhedkar** [comment]
@bnchrch fetched this PR locally, to confirm I need to add a new key to `.env.dev` here and set it the full path of source definitions? (or folder name?)

`LOCAL_CONNECTOR_CATALOG_PATH=/Users/prateekmukhedkar/code/airbyte/airbyte-config/init/src/main/re`sources/seed/source_definitions.yaml`

**Comment by bnchrch** [comment]
@prateekmukhedkar Would you be able to hold off on using this for a couple more hours.

Im running this again this morning and something very strange is going on with the environment variables in docker.

I just want to make sure this will 100% work for you.

**Comment by bnchrch** [comment]
Sorry to be a blocker.

but here are a few steps that should unblock yourself if you need to 

## 1. Generating the OSS Catalog
In the `airbyte` repo run
```sh
SUB_BUILD=ALL_CONNECTORS ./gradlew :airbyte-config:specs:generateOssConnectorCatalog
```

## 2. Move the OSS catalog to platform
In the `airbyte` repo run
```sh
cp airbyte-config/init/src/main/resources/seed/oss_catalog.json <PATH_TO_PLATFORM_REPO>/airbyte-config/init/src/main/resources/seed/oss_catalog.json
```

### 3. Disable catalog download
Comment out these lines in airbyte-platform
https://github.com/airbytehq/airbyte-platform/blob/main/airbyte-config/specs/build.gradle#L15-L45

### 4. Run platform
In the `airbyte-platform` run
```sh
VERSION=dev docker-compose up
```

**Comment by bnchrch** [comment]
closed in favor of https://github.com/airbytehq/airbyte-platform-internal/pull/4705

**Pull request closed without merging**
